pkg/controller/integration/monitor.go

```diff
@@ -57,11 +57,28 @@ func (action *monitorAction) Name() string {
 }
 
 func (action *monitorAction) CanHandle(integration *v1.Integration) bool {
+	// When in InitializationFailed condition a kit is not available for the integration
+	// so the monitor action is not able to handle it.
+	if isInInitializationFailed(integration.Status) {
```

Comment:
How is the integration going to move from error to running phase if it gets updated to fix the issue? My understanding is that the `monitor` action should still be applied to check whether the digest has changed? If that understanding is correct, the `isInInitializationFailed` check should be moved in the `Handle` method.

Reply:
done